numpy/typing/_dtype_like.py

```diff
@@ -79,3 +124,107 @@ class _SupportsDType(Protocol):
 # therefore not included in the Union defining `DTypeLike`.
 #
 # See https://github.com/numpy/numpy/issues/16891 for more details.
+
+# Aliases for commonly used dtype-like objects.
+# Note that the precision of `np.number` subclasses is ignored herein.
+_DTypeLikeBool = Union[
+    Type[bool],
+    Type[np.bool_],
+    "np.dtype[np.bool_]",
+    "_SupportsDType[np.dtype[np.bool_]]",
+    _BoolCodes,
+]
+_DTypeLikeUInt = Union[
+    Type[np.unsignedinteger],
+    "np.dtype[np.unsignedinteger]",
+    "_SupportsDType[np.dtype[np.unsignedinteger]]",
+    _UInt8Codes,
+    _UInt16Codes,
+    _UInt32Codes,
+    _UInt64Codes,
+    _UByteCodes,
+    _UShortCodes,
+    _UIntCCodes,
+    _UIntPCodes,
+    _UIntCodes,
+    _ULongLongCodes,
+]
+_DTypeLikeInt = Union[
+    Type[int],
+    Type[np.signedinteger],
+    "np.dtype[np.signedinteger]",
+    "_SupportsDType[np.dtype[np.signedinteger]]",
+    _UInt8Codes,
+    _Int16Codes,
+    _Int32Codes,
+    _Int64Codes,
+    _ByteCodes,
+    _ShortCodes,
+    _IntCCodes,
+    _IntPCodes,
+    _IntCodes,
+    _LongLongCodes,
+]
+_DTypeLikeFloat = Union[
+    Type[float],
+    Type[np.floating],
+    "np.dtype[np.floating]",
+    "_SupportsDType[np.dtype[np.floating]]",
+    _Float16Codes,
+    _Float32Codes,
+    _Float64Codes,
+    _HalfCodes,
+    _SingleCodes,
+    _DoubleCodes,
+    _LongDoubleCodes,
```

Comment:
A typical example of the newly introduced `_DTypeLike<X>` unions:
* A `np.generic` subclass.
* A builtin counterpart to aforementioned `np.generic` subclass.
* A `np.dtype` instance.
* An object with the `dtype` attribute.
* Various string literals with relevant character codes (`f`, `float`, `f4`, _etc._).